client/state/notices/middleware.js

```diff
@@ -76,6 +78,17 @@ export function onPostSaveSuccess( dispatch, action ) {
 	}
 }
 
+export function onGuidedTransferHostDetailsSaveFailure( dispatch, { error } ) {
+	if ( error && error.message ) {
+		dispatch( errorNotice( error.message ) );
+	} else {
+		dispatch( errorNotice( translate( 'Whoops, there was an error saving your details. Please ' +
```

Comment:
That's a pretty long and not very helpful error. If the error is related directly to the form submission, we can put the notice inside the form—perhaps below the sectionHeader as a block-level element.

Also we should avoid saying things like 'send us a message' if there is no noticeAction that triggers a contact form or chat—the user has no way to actually send a message from here.

Could the message say 'Your hosting details are incorrect. Please try again or [contact support](#)'

The support link could be inline, or if that's the primary action it can be a `noticeAction`